packages/ember-glimmer/tests/integration/components/curly-components-test.js

```diff
@@ -917,4 +920,1044 @@ moduleFor('Components test: curly components', class extends RenderingTest {
+      {{sample-component "one" second="two" id="one-positional"}}
+      {{sample-component first="one" second="two" id="no-positional"}}`);
+
+    assert.equal(this.$('#two-positional').text(), 'one - two');
+    assert.equal(this.$('#one-positional').text(), 'one - two');
+    assert.equal(this.$('#no-positional').text(), 'one - two');
+
+    this.runTask(() => this.rerender());
+
+    assert.equal(this.$('#two-positional').text(), 'one - two');
+    assert.equal(this.$('#one-positional').text(), 'one - two');
+    assert.equal(this.$('#no-positional').text(), 'one - two');
+  }
+
+  ['@skip dynamic arbitrary number of positional parameters'](assert) {
+    this.registerComponent('sample-component', {
+      ComponentClass: Component.extend().reopenClass({
+        positionalParams: 'n'
+      }),
+      template: strip`
+        {{#each n as |name|}}
+          {{name}}
+        {{/each}}`
+    });
+
+    this.render(strip`
+      {{sample-component user1 user2 id="direct"}}
+      {{component "sample-component" user1 user2 id="helper"}}`,
+      {
+        user1: 'Foo',
+        user2: 4
+      }
+    );
+
+    assert.equal(this.$('#direct').text(), 'Foo4', 'direct');
+    assert.equal(this.$('#helper').text(), 'Foo4', 'helper');
+
+    this.runTask(() => this.rerender());
+
+    assert.equal(this.$('#direct').text(), 'Foo4', 'direct');
+    assert.equal(this.$('#helper').text(), 'Foo4', 'helper');
+
+    this.runTask(() => this.context.set('user1', 'Bar'));
+
+    assert.equal(this.$('#direct').text(), 'Bar4', 'direct');
+    assert.equal(this.$('#helper').text(), 'Bar4', 'helper');
+
+    this.runTask(() => this.context.set('user2', '5'));
+
+    assert.equal(this.$('#direct').text(), 'Bar5', 'direct');
+    assert.equal(this.$('#helper').text(), 'Bar5', 'helper');
+
+    this.runTask(() => {
+      this.context.set('user1', 'Foo');
+      this.context.set('user2', 4);
+    });
+
+    assert.equal(this.$('#direct').text(), 'Foo4', 'direct');
+    assert.equal(this.$('#helper').text(), 'Foo4', 'helper');
+  }
+
+  ['@htmlbars with ariaRole specified']() {
+    this.registerComponent('aria-test', {
+      template: 'Here!'
+    });
+
+    this.render('{{aria-test ariaRole=role}}', {
+      role: 'main'
+    });
+
+    this.assertComponentElement(this.firstChild, { attrs: { role: 'main' } });
+
+    this.runTask(() => this.rerender());
+
+    this.assertComponentElement(this.firstChild, { attrs: { role: 'main' } });
+
+    this.runTask(() => this.context.set('role', 'input'));
+
+    this.assertComponentElement(this.firstChild, { attrs: { role: 'input' } });
+
+    this.runTask(() => this.context.set('role', 'main'));
+
+    this.assertComponentElement(this.firstChild, { attrs: { role: 'main' } });
+  }
+
+  ['@htmlbars `template` specified in component is overriden by block']() {
+    this.registerComponent('with-block', {
+      ComponentClass: Component.extend({
+        layout: compile('{{yield}}'),
```

Comment:
I have added a similar version to this with a dynamic value for the name in the block (so we can use UR in the tests). It will be coming in the next push.